intelmq/tests/bots/outputs/smtp_batch/test_output.py

```diff
@@ -0,0 +1,114 @@
+# SPDX-FileCopyrightText: 2022 CSIRT.cz <https://csirt.cz>
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# -*- coding: utf-8 -*-
+from pathlib import Path
+import time
+from types import SimpleNamespace
+import unittest
+
+try:
+    from envelope import Envelope
+except ImportError:
+    from intelmq.lib.exceptions import MissingDependencyError
+    raise MissingDependencyError('envelope', '>=2.0.0')
+
+import intelmq.lib.test as test
+from intelmq.bots.outputs.smtp_batch.output import SMTPBatchOutputBot
+from intelmq.lib.cache import Cache
+
+SENT_MESSAGES = []  # here we collect e-mail messages there were to be sent
+BOT_ID = "test-bot"
+IDENTITY1 = 'one@example.com'
+KEY1 = f"{BOT_ID}:{IDENTITY1}".encode()
+EVENT1 = {'__type': 'Event',
+          'source.ip': '127.0.0.1',
+          'source.url': 'http://example.com/',
+          'source.abuse_contact': IDENTITY1
+          }
+IDENTITY2 = 'one@example2.com'
+KEY2 = f"{BOT_ID}:{IDENTITY2}".encode()
+EVENT2 = {'__type': 'Event',
+          'source.ip': '127.0.0.2',
+          'source.url': 'http://example2.com/',
+          'source.abuse_contact': IDENTITY2
+          }
+
+REDIS_LOGIN = {"redis_cache_host": "127.0.0.1",
+               "redis_cache_port": 6379,
+               "redis_cache_db": "14",
+               "redis_cache_ttl": ""}
+MAIL_TEMPLATE = Path("mail_template.txt")
+FROM_IDENTITY = "from-example@example.com"
+
+
+class TestSMTPBatchOutputBot(test.BotTestCase, unittest.TestCase):
+
+    @classmethod
+    def set_bot(cls):
+        cls.bot_reference = SMTPBatchOutputBot
+        cls.sysconfig = {"alternative_mails": "",
+                         "attachment_name": "events_%Y-%m-%d",
+                         "bcc": [],
+                         "email_from": FROM_IDENTITY,
+                         "gpg_key": "",
+                         "gpg_pass": "",
```

Comment:
I've found the key that could be added with `"gpg_key": Path(__file__).parent / "../../../assets/key-private.pgp"`.
But since we replace the `envelope.Envelope.send` method of the 3rd party library, we skip any GPG processing.